src/bfuse16.rs

```diff
@@ -77,27 +77,42 @@ impl Filter<u64> for BinaryFuse16 {
     }
 }
 
+impl BinaryFuse16 {
+    /// Try to construct the filter from a key iterator. Can be used directly
+    /// if you don't have a contiguous array of u64 keys.
+    ///
+    /// Note: the iterator will be iterated over multiple times while building
+    /// the filter. If using a hash function to map the key, it may be cheaper
+    /// just to create a scratch array of hashed keys that you pass in.
+    fn try_from_iterator<'a, T>(keys: T) -> Result<Self, &'static str>
+    where
+        T: ExactSizeIterator<Item = &'a u64> + Clone,
+    {
+        bfuse_from_impl!(keys fingerprint u16, max iter 1_000)
+    }
+}
```

Comment:
What do you think of implementing

```
impl<'a, I : ExactSizeIterator<Item = &'a u64> + Clone> TryFrom<I> for BinaryFuse16 {
  ..
}
```

here instead?

Reply:
I had tried very hard to get the trait to work but I couldn't figure it out. I don't know if I tried your specific snippet but for me it's failing with:
```
error[E0119]: conflicting implementations of trait `core::convert::TryFrom<_>` for type `bfuse32::BinaryFuse32`
  --> src/bfuse32.rs:81:1
   |
81 | impl <'a, I: ExactSizeIterator<Item = &'a u64> + Clone> TryFrom<I> for BinaryFuse32 {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `core`:
           - impl<T, U> TryFrom<U> for T
             where U: Into<T>;

```

I'm a Rust novice so maybe the fix is obvious to you.

Reply:
I see. Yeah, that makes sense - looks good as is. Thanks